File: ScriptBeeClient/src/app/services/plugin/marketplace-plugin.ts
import { PluginKind } from "./plugin";

export interface MarketplacePlugin {
  id: string;
  name: string;
  authors: string[];
  description: string;
  versions: PluginVersion[];
}

export interface PluginVersion {
  version: string;
  extensionPointVersions: ExtensionPointVersion[];
  installed: boolean;
}

export interface ExtensionPointVersion {
  kind: PluginKind;
  version: string;
}

File: ScriptBeeWebApp/Controllers/DTO/MarketplacePlugin.cs
using System.Collections.Generic;

namespace ScriptBeeWebApp.Controllers.DTO;

public sealed record MarketplacePlugin(
    string Id,
    string Name,
    string Description,
    List<string> Authors,
    List<PluginVersion> Versions
);

public sealed record PluginVersion(string Version, List<ExtensionPointVersion> ExtensionPointVersions, bool Installed);

public sealed record ExtensionPointVersion(string Kind, string Version);
